lib/src/ballon.dart

```diff
@@ -2,75 +2,67 @@ part of tooltip;
 
 class _Ballon extends StatefulWidget {
   final TooltipDirection tooltipDirection;
-  final Offset targetCenter;
+  final Offset? targetCenter;
   final double borderRadius;
   final double arrowBaseWidth;
   final double arrowTipDistance;
   final Color borderColor;
   final double borderWidth;
-  // final double left;
-  // final double top;
-  // final double right;
-  // final double bottom;
   final double arrowLength;
   final Widget content;
   final EdgeInsets ballonPadding;
   final Color backgroundColor;
   final List<BoxShadow> shadows;
-  final Function onTap;
+  final GestureTapCallback? onTap;
   final Function(_BallonSize) onSizeChange;
 
   const _Ballon({
-    Key key,
-    // this.left,
-    // this.top,
-    // this.right,
-    // this.bottom,
-    @required this.tooltipDirection,
-    @required this.borderRadius,
-    @required this.arrowBaseWidth,
-    @required this.arrowTipDistance,
-    @required this.borderColor,
-    @required this.borderWidth,
-    @required this.arrowLength,
-    @required this.content,
-    @required this.ballonPadding,
-    @required this.backgroundColor,
-    @required this.shadows,
+    Key? key,
+    required this.tooltipDirection,
+    required this.borderRadius,
+    required this.arrowBaseWidth,
+    required this.arrowTipDistance,
+    required this.borderColor,
+    required this.borderWidth,
+    required this.arrowLength,
+    required this.content,
+    required this.ballonPadding,
+    required this.backgroundColor,
+    required this.shadows,
     this.targetCenter,
     this.onTap,
-    @required this.onSizeChange,
+    required this.onSizeChange,
   }) : super(key: key);
 
   @override
-  __BallonState createState() => __BallonState();
+  _BallonState createState() => _BallonState();
 }
 
-class __BallonState extends State<_Ballon> {
-  _BallonSize _lastSizeNotified;
+class _BallonState extends State<_Ballon> {
+  _BallonSize? _lastSizeNotified;
 
   GlobalKey _containerKey = GlobalKey();
 
   @override
   Widget build(BuildContext context) {
-    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+    WidgetsBinding.instance!.addPostFrameCallback((timeStamp) {
       if (!mounted) return;
-      final RenderBox renderBox =
-          _containerKey.currentContext.findRenderObject();
-      final Size size = renderBox.size;
+      final RenderBox? renderBox =
+          _containerKey.currentContext?.findRenderObject() as RenderBox?;
       if (renderBox == null) return null;
```

Comment:
Does return null make sense here or just return like when we do the `mounted` check?

Reply:
Hmmmm strange apparently Dart allows for return or return `null`; seems it does the same thing. To stay consistent, I think the best option is to remove the null. We can actually also add this to our linter: https://dart-lang.github.io/linter/lints/avoid_returning_null_for_void.html